src/mats_l1_processing/read_parquet_functions.py

```diff
@@ -185,6 +185,7 @@ def read_ccd_data_in_interval(
     stop: datetime,
     path: str,
     filesystem: Optional[pa.fs.FileSystem] = None,
+    filter: dict = None,
```

Comment:
The variable is wrongly typed! The important part here is the `Optional`, but I think it would be better to type it completely:
```suggestion
    filter: Optional[Dict[str, Sequence[float, float]]] = None,
```

Reply:
This will require that you import `Sequence` from `typing` up top. An alternative is to use `List` instead of `Sequence` (the latter is a little more general, and more appropriate I think, but either works).